README.md

```diff
@@ -65,146 +65,235 @@ TechInfo lookup is looking at Toyota's Techinfo site (payment required, minimum
+
+1-4. Turn on your Comma 3X by plugging in the right-angled USB-C cable to the harness.
+
+1-5. Connect it to your Wi-Fi network.
+
+1-6. Install "Custom Software" with URL `commaai/master`
+
+### Step 2. SSH into the device
+
+2-1. Before you start (for both macOS and Windows): https://github.com/commaai/openpilot/wiki/SSH#before-you-start
+
+2-2-1. macOS: https://github.com/commaai/openpilot/wiki/SSH#option-2mac---pre-installed-openssh-client-on-macos
+
+2-2-2. Windows: https://github.com/commaai/openpilot/wiki/SSH#option-2---pre-installed-openssh-client-on-windows-10-and-up
+
+2-3. SSH into the device.
+```sh
+ssh comma@"your Comma IP"
+```
+
+### Step 3. Extract security key
+
+> [!TIP]
+> If you already have the security key, proceed to Step 5. 
+
+3-1. Put the car into `Not Ready to Drive` mode.
+* Slowly press the `POWER` button twice WITHOUT pressing the brake pedal.
+* The first press turns on `ACCESSORY` mode. The second press activates `Not Ready To Drive` mode shown below.
+  ![PXL_20240718_234619671 MP](https://github.com/user-attachments/assets/4970e82e-e7df-471f-9896-ba532509793d)
+
+3-2. Kill the openpilot process.
+```sh
+pkill -f openpilot
+```
+
+Comma should display just the splash screen with the Comma logo.
+
+3-3. Clone Willem's secoc Git repository.
+```sh
+git clone https://github.com/I-CAN-hack/secoc
+```
+
+3-4. Navigate to the secoc directory.
+```sh
+cd /data/openpilot/secoc
+```
+
+3-5. Run the key extraction script.
+```sh
+./extract_keys.py
+```
+
+3-6. If you see something like this, the key extraction was successful.
+```sh
+comma@comma-71b93b83:/data/openpilot/secoc$ ./extract_keys.py 
+INFO: connecting to panda 2c0004004450383632311333
+Getting application versions...
+ - APPLICATION_SOFTWARE_IDENTIFICATION (application) b'\x018965B4509100\x00\x00\x00\x00'
+ - APPLICATION_SOFTWARE_IDENTIFICATION (bootloader)  b'\x01!!!!!!!!!!!!!!!!'
+
+Security Access...
+ - SEED: 36552fe27172c99222eec3a9b9bd1f28
+ - KEY: b7b55ba16369bba912b7aa4c06e6c35e
+ - Key OK!
+
+Preparing to upload payload...
+ - Write data by identifier 0x201 00000000000000000000000000000000
+ - Write data by identifier 0x202 00000000000000000000000000000000
+
+Upload payload...
+ - Request download
+ - Transfer data 0
+ - Transfer data 1
+ - Transfer data 2
+ - Transfer data 3
+
+Verify payload...
+ - Routine control 0x10f0 OK!
+
+Trigger payload...
+
+Dumping keys...
+100%|████████████████████████| 448/448 [00:00<00:00, 14293.36it/s]
+
+ECU_MASTER_KEY    82667ef509b9f07a134aaf89d4973c68
+SecOC Key (KEY_4) 0123456789abcdef0123456789abcdef
+
+SecOC key written to param successfully!
+comma@comma-71b93b83:/data/openpilot/secoc$ 
+```
+
+This 32 digit hexadecimal number is your key.
+> SecOC Key (KEY_4) **0123456789abcdef0123456789abcdef**
+
+Email yourself the key so that you don't need to extract it again.
```

Comment:
Some people don't store every email so maybe this should be a "Archive and keep your key safe. Perhaps email yourself". People might have a different process for archival.

Reply:
LGTM with rewording